chia/consensus/block_body_validation.py

```diff
@@ -373,7 +378,8 @@ async def validate_block_body(
             unspent = removal_coin_records[npc.coin_name]
             error = blockchain_check_conditions_dict(
                 unspent,
-                announcements,
+                coin_announcement_names,
+                puzzle_announcement_names,
```

Comment:
Why do we need these when we have the `condition_dict`?

Reply:
Ahhh, I see... the `condition_dict` doesn't have the coin its associated with. Well, we really need to fix that.